### ISSUE TITLE
Multiple CPIDs in Hosts

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#557]
**Reported by KAMCOBILL on 10 May 38204111 10:13 UTC**
Hi,

I filling this ticket out as best as I can to explain the problem. I have multiple CPIDs in each host (15 each). I started with BOINC somewhere in Feb 2005, and around Nov, 2006 when I went from 5.4.11 to the next the CPID problem started. My Username, Password, Email Address, URL, Country, Zip Code, Team Name, Team Founder are all the same at every project (BOINC Managed and unmanaged. In the beginning I attached 4 hosts to all the projects and as I added host and projects all 14 nodes are attached to all the projects. During the time that I went from 4 to 15 hosts I've had as many as 10 CPIDs and No Less than 4, which I have currently. I tried to attached the last host(15th) one project at a time, using another host to change the CPID at that project to my main CPID. I still ended up 4 CPIDs in the new host. I think other hosts may have connected to server between my update and attachment. My Host seesaw the CPIDs back and forth when a differet hosts connects to server. 

To make a long story short and not having to detach the hosts from the projects that have the wrong CPID in it one at a time and uninstalling BOINC Manager & deleting BOINC Directory on all 15 hosts, is there a way you can put a feature in the tools tab to synchronize CPIDs/Hash in the hosts to the main one (Something like BOINCStats Synchronize Retired Projects)? Or someway to use BAM to synchronize the CPIDs? I've exhausted everything I can think of to stop my hosts from controlling the CPID at the projects with no luck.

Thanks

Bill 

Migrated-From: http://boinc.berkeley.edu/trac/ticket/579

**Comment by romw** [comment]
**Commented by Nicolas on 3 Jan 38204154 03:06 UTC**
Do you have all hosts attached to all projects? I think that would sync CPIDs. Additionally, you have to merge duplicates on each project website.

**Comment by romw** [comment]
**Commented by Nicolas on 7 Nov 38204160 03:06 UTC**
Wait, scratch that completely. I thought you were talking about [host CPIDs](CrossProjectId#hosts), not about different [user CPIDs](CrossProjectId#users) on each host.

**Comment by romw** [comment]
**Commented by KAMCOBILL on 7 Nov 38205889 18:13 UTC**
Replying to [Ageless](comment:1):
Thanks Ageless,

I didn't know what to put in them.

Bill

**Comment by romw** [comment]
**Commented by KAMCOBILL on 13 Nov 38205908 04:53 UTC**
Replying to [Nicolas](comment:3):
> Wait, scratch that completely. I thought you were talking about [host CPIDs](CrossProjectId#hosts), not about different [user CPIDs](CrossProjectId#users) on each host.

I have 4 CPID in all 15 hosts and they change the CPIDs back and forth when they connect. Each host has different CPIDs for the same project and each project changes to whatever CPID a perticular host has in it. As far as I can see the host CPIDs has a hash CPIDs (not sure what that is) that has something to do with User CPID. Anyway. All host have been or are attached to all projects, except Host 15 and it's
only connected to 61 projects and waiting to attach to down project and unreachable.

**Comment by romw** [comment]
**Commented by KAMCOBILL on 30 Dec 38205933 10:40 UTC**
Replying to [Nicolas](comment:2):
> Do you have all hosts attached to all projects? I think that would sync CPIDs. Additionally, you have to merge duplicates on each project website.

Sorry, I clicked to the wrong repy button. I have merged all host at projects that will let you. I should not have any more than 17 hosts. My BOINCStats has 59 at this time. Some projects has the smae host name and description and won't let you merge them for some reason. The multiple host doesn't really cause as big as problem for overall user stats as the CPID (e.g. User stats at BoincStats range from near zero to over 600K+ for my user stats. Here's the 30 for me to show how they go up and down: http://boincstats.com/stats/boinc_user_graph.php?pr=bo&id=b65b36154e3a49f636a3148d4e0b3c83 <Main CPID
